test/Microsoft.AspNetCore.Authentication.Test/CookieTests.cs

```diff
@@ -370,11 +370,40 @@ private Task SignOutAsWrong(HttpContext context)
                 o.SlidingExpiration = false;
                 o.Events = new CookieAuthenticationEvents
                 {
-                    OnValidatePrincipal = ctx =>
+                    OnValidatePrincipal = async ctx =>
                     {
-                        ctx.RejectPrincipal();
-                        ctx.HttpContext.SignOutAsync("Cookies");
-                        return Task.FromResult(0);
+                        ctx.Fail("Authentication was aborted from user code.");
+                        await ctx.HttpContext.SignOutAsync("Cookies");
+                    }
+                };
+            },
+            context =>
+                context.SignInAsync("Cookies",
+                    new ClaimsPrincipal(new ClaimsIdentity(new GenericIdentity("Alice", "Cookies")))));
+
+            var transaction1 = await SendAsync(server, "http://example.com/testpath");
+
+            var exception = await Assert.ThrowsAsync<Exception>(delegate
+            {
+                return SendAsync(server, "http://example.com/checkforerrors", transaction1.CookieNameValue);
+            });
+
+            Assert.Equal("Authentication was aborted from user code.", exception.InnerException.Message);
+        }
+
+        [Fact]
+        public async Task CookieCanBeSkippedAndSignedOutByValidator()
+        {
+            var server = CreateServer(o =>
+            {
+                o.ExpireTimeSpan = TimeSpan.FromMinutes(10);
+                o.SlidingExpiration = false;
+                o.Events = new CookieAuthenticationEvents
+                {
+                    OnValidatePrincipal = async ctx =>
+                    {
+                        ctx.None();
```

Comment:
Removing this test now, as there's no skip concept, you can manipulate only the principal from ValidateContext, no control flow to deal with...